tinygrad/tensor.py

```diff
@@ -544,13 +538,13 @@ def tan(self): return self.sin() / self.cos()
 
   @staticmethod
   def _tri(r:int, c:int, k:int=0, **kwargs) -> Tensor: return Tensor.arange(r, **kwargs).unsqueeze(1).expand(r,c) <= Tensor.arange(-k, c-k, **kwargs).unsqueeze(0).expand(r,c)
-  def triu(self, k:int=0) -> Tensor: return Tensor._tri(self.shape[-2], self.shape[-1], k=k, dtype=self.dtype).where(self, Tensor.zeros_like(self))
-  def tril(self, k:int=0) -> Tensor: return Tensor._tri(self.shape[-2], self.shape[-1], k=k+1, dtype=self.dtype).where(Tensor.zeros_like(self), self)
+  def triu(self, k:int=0) -> Tensor: return Tensor._tri(self.shape[-2], self.shape[-1], k=k, dtype=self.dtype) * self
```

Comment:
Err, there's issues with inf for this

Reply:
Currently it supports inf but is it a bug? 
Tensor(float("inf")) * 0 = nan
Tensor([float("inf")]) * Tensor(0) = 0
Tensor([float("inf")]) * Tensor([0]) = nan
Can we trust this behaviour?

Reply:
i knew there was a reason it was where, where has better inf and nan handing than a masked multiple does

Reply:
@wozeparrot So can we use .where for "def pad" as well. We rely on 1/inf = 0